src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -491,4 +495,202 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
         assertEquals(1, checkouts.size());
         assertEquals(GitSCM.class, checkouts.get(0).getClass());
     }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Basic test to prove that checkout gets turned off correctly
+    public void baselineResetSingleRepo() throws Exception {
+        sampleRepo.init();
+        TaskListener listener = StreamTaskListener.fromStdout();
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("semaphore 'waitFirst'\n" +
+                "node {\n" +
+                "    checkout(changelog:" + true +", poll:" + true + ", scm: [$class: 'GitSCM', branches: [[name: '*/master']], " +
+                "doGenerateSubmoduleConfigurations: false, extensions: [], " +
+                "gitTool: 'Default', submoduleCfg: [], userRemoteConfigs: [[url: '" + sampleRepo + "']]])" +
+                "}\n", false /* for org.jvnet.hudson.test.FakeChangeLogSCM */));
+
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+
+        SemaphoreStep.waitForStart("waitFirst/1", b1);
```

Comment:
Is this redundant given `.waitForStart()` above?

Reply:
They have different meanings (the first one waits for the job to enter the queue and start running, the second waits for the corresponding `semaphore` step in the Pipeline to start executing), but in this case, since the test doesn't do or check anything between `SemaphoreStep.waitForStart` and `SemaphoreStep.success`, the semaphore step is redundant and it could just be removed, but so is the `.waitForStart()` above, since the next line just waits for the build to complete. This should be all that is needed:

```
WorkflowRun b1 = r.buildAndAssertSuccess(p);
```